src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -874,7 +874,7 @@ private boolean parseVariables( final Type t, final BasicScope parentScope )
 		}
 	}
 
-	private Variable parseVariable( final Type t, final BasicScope scope )
+	private Variable parseVariable( final Type t, final BasicScope scope, final boolean allowInitialization )
 	{
```

Comment:
Rereading this PR.

All of this seems unnecessarily complicated, since this function is handling two separate cases, controlled by the boolean parameter. It's made worse now that you have to track this state throughout the function, as opposed to in a single place.

What we should do instead is split this into parseDeclaration and parseDefinition (or repurpose parseAssignment to optionally only allow "="), then move the "cannot initialize parameter" complaint into something that actually deals with function parameters.

(or, if you want different names, parseVariable and parseInitialization.)

Reply:
In particular, I might suggest as API:

Variable parseVariable(Type t) { ... }
Assignment parseInitialization(Variable v) {
  // parses " = ...;"
}

then add it to the scope in question parseVariables. That way, if you really want to parse initialization of a variable in a parameter declaration, you can do that without impacting the scope.

Reply:
(do you get notified if there are commits after you posted a comment, or do you need to be notified manually?)

Reply:
(I do, I just haven't had time to look at this.)